zcash_client_backend/src/data_api/chain.rs

```diff
@@ -169,74 +149,60 @@ where
     );
 
     // Start at either the provided height, or where we synced up to previously.
-    let (scan_from, commitment_tree_meta) = from_height.map_or_else(
-        || {
-            data_db.fully_scanned_height().map_or_else(
-                |e| Err(Error::Wallet(e)),
-                |last_scanned| {
-                    Ok(last_scanned.map_or_else(|| (None, None), |(h, m)| (Some(h + 1), Some(m))))
+    let (scan_from, mut prior_block_metadata) = match from_height {
+        Some(h) => {
+            // if we are provided with a starting height, obtain the metadata for the previous
+            // block (if any is available)
+            (
+                Some(h),
+                if h > BlockHeight::from(0) {
+                    data_db.block_metadata(h - 1).map_err(Error::Wallet)?
+                } else {
+                    None
                 },
             )
-        },
-        |h| Ok((Some(h), None)),
-    )?;
+        }
+        None => {
+            let last_scanned = data_db.block_fully_scanned().map_err(Error::Wallet)?;
+            last_scanned.map_or_else(|| (None, None), |m| (Some(m.block_height + 1), Some(m)))
+        }
+    };
 
-    block_source.with_blocks::<_, DbT::Error, DbT::NoteRef>(
-        scan_from,
-        limit,
-        |block: CompactBlock| {
-            add_block_to_runner(params, block, &mut batch_runner);
-            Ok(())
-        },
-    )?;
+    block_source.with_blocks::<_, DbT::Error>(scan_from, limit, |block: CompactBlock| {
+        add_block_to_runner(params, block, &mut batch_runner);
```

Comment:
It occurs to me that we should actually do the continuity check here, because there's no reason to add a block to the runner or continue the scan if the continuity check fails.

Reply:
I've made this change in the forthcoming range-suggestions PR.